drivers/include/hmc5883l.h

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_hmc5883l HMC5883L 3-axis digital compass
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Device driver for the Honeywell HMC5883L 3-axis digital compass
+ *
+ * The driver implements basic polling mode. The application can use
+ * different approaches to get new data, either
+ *
+ * - using the #hmc5883l_read function at a lower rate than the the DOR, or
+ * - using the data-ready interrupt (**DRDY**), see #hmc5883l_init_int.
+ *
+ * The data-ready interrupt (**DRDY**) is only be available when module
+ * `hmc5883l_int` is enabled.
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ *
+ * @{
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#ifndef HMC5883L_H
+#define HMC5883L_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+/** HMC5883L I2C address*/
+#define HMC5883L_I2C_ADDRESS    (0x1e)
+
+/** HMC5883L chip id defined in Identification Registers A..C */
+#define HMC5883L_ID             { 0x48, 0x34, 0x33 }
+
+/** HMC5883L chip id length */
+#define HMC5883L_ID_LEN         (3)
+
+/** Definition of error codes */
+typedef enum {
+    HMC5883L_OK,                   /**< success */
+    HMC5883L_ERROR_I2C,            /**< any I2C communication error */
```

Comment:
to make the error checking less error prone, I think it is a good idea to assign them to negative values in the enum definition here (e.g. `HMC5883L_ERROR_I2C = -1` ...). Right now, the code like the following would yield unexpected behavior:
```c
int res = hmc5883l_init(...);
if (res == HMC5883L_ERROR_I2C) {   // fail: minus sign is missing
    // do some specific error handling
```
Assigning these enum values would further make this code consistent to other device drivers in RIOT.

Reply:
Yeah, that's a problem. When I started to port some drivers I already used in other projects to RIOT, I was trying to figure out, what the right way in RIOT is to define error codes. I had to realize that all approaches are used:

1. negative codes that are returned directly (for example `drivers/bmx280`)
2. positive codes that are returned inverted (for example `drivers/bmp180`)
3. positive codes that are returned directly.

Even for same kind of drivers, they differ. Most of the drivers use 1. and 2.

Therefore, I decided to define and use them like standard error codes, i.e., positive codes returned inverted.

But, I can change it if it is really necessary. The documentation refers negative values.

Reply:
I changed it as proposed.